### ISSUE TITLE
Inquiry Regarding Fullscreen Mode in VirtualDub2

### ISSUE THREAD

**Issue by felipemlx** [shekh/VirtualDub2#4]
Dear @shekh ,
Hope you are doing well!!

I hope this message finds you well. First of all, I want to thank you for the excellent work on VirtualDub2—it's truly a fantastic tool that meets all my needs.

I'm reaching out because I've encountered a small issue with the fullscreen mode in VirtualDub2. When I press Alt + Enter to enter fullscreen mode, it works perfectly. However, as soon as I switch to another program, VirtualDub2 automatically exits fullscreen mode.

Would there be any existing option within the program to lock or maintain fullscreen mode even when switching to another application? If not, would you be able to provide some guidance on how I might modify the source code to achieve this? I would like to build a version of VirtualDub2 that keeps the application in fullscreen mode, regardless of focus changes.

Thank you very much for your time and assistance. I look forward to hearing from you.

**Comment by shekh** [comment]
Thanks. I'm not sure, this should work. First, check out these options: https://sourceforge.net/p/vdfiltermod/wiki/display/

If you still want to dig into source, I strongly advice to focus on d3d9 mode only (there is support for other display methods).
You can start searching from display.cpp, displaydrv3d.cpp, direct3d.cpp
Also check what is going on in displaymgr.cpp on WM_ACTIVATEAPP message.

**Comment by felipemlx** [comment]
Thanks for the quick reply! I made some changes to the code but I'm having trouble building it. I've tried to follow the BUILDING.txt tutorial but still couldn't do it, I'm having several problems with YASM. Could you tell me if the BUILDING.txt tutorial is up to date and working? Thanks!

**Comment by shekh** [comment]
Yes, BUILDING.txt should be good. Do you have any message output from yasm?

**Comment by felipemlx** [comment]
First, I downloaded Visual Studio 2015 Community with Windows XP Support for C++. Then, I downloaded Visual Studio Update 3. After that, I installed the DirectX SDK from June 2010.

As for YASM, I downloaded the Win64.exe and placed it in the folder C:\Program Files (x86)\VC\bin.

Regarding the project, I opened the Virtual-Dub-vc14.slm solution.

I had to change the Platform Toolset to Visual Studio 2015 (v140) because the v141_xp toolset was missing and I could not find a way to download it. I left the Target Platform Version at 8.1 (no changes). With all these settings, I tried to build a Release Win32. Here are the errors I received:

Error	MSB3721	The command "yasm -X vc -g cv8 -f win32 -o "..\..\obj\Release\system\a_memory.obj" "C:\Users\User\Desktop\VirtualDub2-master\src\system\source\a_memory.asm"" exited with code 1.	
Project: system	
File: C:\Users\User\Desktop\VirtualDub2-master\src\YASM.targets	line 46	

Error	MSB3721	The command "yasm -X vc -g cv8 -f win32 -o "..\..\obj\Release\Meia\a_predict_isse.obj" "C:\Users\User\Desktop\VirtualDub2-master\src\Meia\source\a_predict_isse.asm"" exited with code 1.	
Project: Meia	
File: C:\Users\User\Desktop\VirtualDub2-master\src\YASM.targets	line 46	

Error	MSB3721	The command ""..\..\out\Release\asuka.exe" psa "C:\Users\User\Desktop\VirtualDub2-master\src\VDFilters\source\VFBlendLayers.vdshaders" "C:\Users\User\Desktop\VirtualDub2-master\src\VDFilters\\autogen\\VFBlendLayers.inl"" exited with code 1.	
Project: VDFilters	
File: C:\Users\User\Desktop\VirtualDub2-master\src\PSA.targets	line 46	

Error	LNK1181	cannot open input file 'system.lib'	
Project: Asuka	
File: C:\Users\User\Desktop\VirtualDub2-master\src\Asuka\LINK	line 1

Error	LNK1181	cannot open input file 'system.lib'	
Project: test	
File: C:\Users\User\Desktop\VirtualDub2-master\src\test\LINK	line 1	

Error	LNK1181	cannot open input file 'C:\Users\User\Desktop\VirtualDub2-master\lib\Release\system.lib'	
Project: Ami	
File: C:\Users\User\Desktop\VirtualDub2-master\src\Ami\LINK	line 1	

Error	LNK1181	cannot open input file 'C:\Users\User\Desktop\VirtualDub2-master\lib\Release\system.lib'	
Project: disasm	
File: C:\Users\User\Desktop\VirtualDub2-master\src\disasm\LINK	line 1	

Error	MSB6006	"cmd.exe" exited with code 9009.	
Project: Kasumi	
File: C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets	line 171	

Error	MSB6006	"cmd.exe" exited with code 9009.	
Project:Riza	
File: C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets	line 171	

Error	MSB6006	"cmd.exe" exited with code 9009.	
Project: VDCapture	
File: C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets	line 171	

Error	MSB6006	"cmd.exe" exited with code 9009.	
Project: VDDisplay	
File: C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets	line 171	

Error	MSB6006	"cmd.exe" exited with code 9009.	
Project: VirtualDub	
File: C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets	line 171

**Comment by shekh** [comment]
No idea what is wrong. Try to run yasm alone, e.g.
yasm.exe -X vc -g cv8 -f win32 a_memory.asm

**Comment by felipemlx** [comment]
I finally managed to get it working. The build issues were indeed due to YASM, and the problem was a typo in the PATH. Regarding the files, I made a small change to the SetFullScreen method in captureui.cpp and it worked; I got what I needed. After fixing the typo, the build worked perfectly. I’m immensely grateful for the quick support. And thank you once again for providing such an amazing piece of software!

**Comment by shekh** [comment]
Great!
